src/Native/Unix/configure.cmake

```diff
@@ -362,6 +362,34 @@ check_prototype_definition(
 
 check_cxx_source_compiles(
     "
+    #include <stdlib.h>
+
+    int main()
+    {
+        char* path = strdup(\"abc\");
+        int ret = mkstemps(path, 3);
```

Comment:
Ugh.  In general we like to try to keep these as simple as possible, focused as much as possible on the one function being tested.  If that's not possible in this case because the compiler is being too strict, ok.

You could simplify them slightly by just returning the result of mkstemp{s} rather than returning 0 on a separate line.  The return code doesn't matter as the code isn't going to run, anyway.